spec_util/meld.go

```diff
@@ -602,6 +616,18 @@ func stripOptional(data *pb.Data) *pb.Data {
 	return optional.GetData()
 }
 
+// Converts a singleton one-of into its underlying variant. Does nothing if the
+// given IR node is not a singleton one-of.
+func rewriteSingletonOneOf(data *pb.Data) {
```

Comment:
optional suggestion - a pattern I like for functions that might have no effect is to name them `maybeDoThing`, in this case `maybeRewriteSingletonOneOf`. This makes it super clear at call sites that the function may have no effect.